nsxt/resource_nsxt_lb_icmp_monitor_test.go

```diff
@@ -0,0 +1,168 @@
+					resource.TestCheckResourceAttr(testResourceName, "data_length", dataLength),
+					resource.TestCheckResourceAttr(testResourceName, "tag.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceNsxtLbIcmpMonitor_importBasic(t *testing.T) {
+	name := "test-nsx-monitor"
+	testResourceName := "nsxt_lb_icmp_monitor.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNSXLbIcmpMonitorCheckDestroy(state, name)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNSXLbIcmpMonitorCreateTemplateTrivial(name),
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNSXLbIcmpMonitorExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		nsxClient := testAccProvider.Meta().(*nsxt.APIClient)
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("NSX LB icmp monitor resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("NSX LB icmp monitor resource ID not set in resources ")
+		}
+
+		monitor, responseCode, err := nsxClient.ServicesApi.ReadLoadBalancerIcmpMonitor(nsxClient.Context, resourceID)
+		if err != nil {
+			return fmt.Errorf("Error while retrieving LB icmp monitor with ID %s. Error: %v", resourceID, err)
+		}
+
+		if responseCode.StatusCode != http.StatusOK {
+			return fmt.Errorf("Error while checking if LB icmp monitor %s exists. HTTP return code was %d", resourceID, responseCode.StatusCode)
+		}
+
+		if displayName == monitor.DisplayName {
+			return nil
+		}
+		return fmt.Errorf("NSX LB icmp monitor %s wasn't found", displayName)
+	}
+}
+
+func testAccNSXLbIcmpMonitorCheckDestroy(state *terraform.State, displayName string) error {
+	nsxClient := testAccProvider.Meta().(*nsxt.APIClient)
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_lb_icmp_monitor" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		monitor, responseCode, err := nsxClient.ServicesApi.ReadLoadBalancerIcmpMonitor(nsxClient.Context, resourceID)
+		if err != nil {
+			if responseCode.StatusCode != http.StatusOK {
+				return nil
+			}
+			return fmt.Errorf("Error while retrieving LB icmp monitor with ID %s. Error: %v", resourceID, err)
+		}
+
+		if displayName == monitor.DisplayName {
+			return fmt.Errorf("NSX LB icmp monitor %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNSXLbIcmpMonitorCreateTemplate(name string, count string, interval string, port string, timeout string, dataLength string) string {
+	return fmt.Sprintf(`
+resource "nsxt_lb_icmp_monitor" "test" {
+  description       = "test description"
```

Comment:
I think the spacing in the templates are wrong. As per comments I got for previous resources.
You should make the configuration examples go through terraform fmt first.

Reply:
Will do in next commit

Reply:
Done